src/pybind/mgr/nfs/export.py

```diff
@@ -496,46 +518,44 @@ def get_export_by_pseudo(
         export = self._fetch_export(cluster_id, pseudo_path)
         return export.to_dict() if export else None
 
-    def apply_export(self, cluster_id: str, export_config: str) -> Tuple[int, str, str]:
+    def apply_export(self, cluster_id: str, export_config: str) -> AppliedExportResults:
         try:
-            if not export_config:
-                raise NFSInvalidOperation("Empty Config!!")
-            try:
-                j = json.loads(export_config)
-            except ValueError:
-                # okay, not JSON.  is it an EXPORT block?
+            aeresults = AppliedExportResults()
+            for export in self._read_export_config(cluster_id, export_config):
                 try:
-                    blocks = GaneshaConfParser(export_config).parse()
-                    exports = [
-                        Export.from_export_block(block, cluster_id)
-                        for block in blocks
-                    ]
-                    j = [export.to_dict() for export in exports]
+                    chg = self._apply_export(cluster_id, export)
                 except Exception as ex:
-                    raise NFSInvalidOperation(f"Input must be JSON or a ganesha EXPORT block: {ex}")
-
-            # check export type
-            if isinstance(j, list):
-                ret, out, err = (0, '', '')
-                for export in j:
-                    try:
-                        r, o, e = self._apply_export(cluster_id, export)
-                    except Exception as ex:
-                        r, o, e = exception_handler(ex, f'Failed to apply export: {ex}')
-                        if r:
-                            ret = r
-                    if o:
-                        out += o + '\n'
-                    if e:
-                        err += e + '\n'
-                return ret, out, err
-            else:
-                r, o, e = self._apply_export(cluster_id, j)
-                return r, o, e
+                    msg = f'Failed to apply export: {ex}'
+                    log.exception(msg)
+                    chg = {"state": "error", "msg": msg}
+                aeresults.append(chg)
+            return aeresults
         except NotImplementedError:
-            return 0, " Manual Restart of NFS PODS required for successful update of exports", ""
+            raise ManualRestartRequired("Changes applied")
```

Comment:
Oh, I had some issues with this because it's making what appear to be k8s specific instructions when you get a generic  NotImplementError - which I assume the original author thought would only be raised by the rook module (maybe). I was bothered by the fact that the "error message" indicated nothing about the partial application of the settings - but only that pods may need to be restarted [1]. So the actual text printed will be something like: "Changes applied (Manual Restart of NFS PODS required)"

If you don't like the "Changes applied" bit. I can just drop it and leave the emitted text as-is.

[1] - Pods is not an acronym/initialism, so PODS in all caps is wrong IMO, but I didn't change that despite it bugging me.

Reply:
Ah! This makes sense. 

Let's make s/PODS/pods/ change in this PR or in a follow up PR.

Reply:
Changed the spelling to "Pods".